### ISSUE TITLE
No EMS yet for host with extracted Host-ID

### ISSUE THREAD

**Issue by mf3d3r** [openbaton/NFVO#198]
Hi @gc4rella,
thank you for answering in the other thread. 

I'm still trying to get up the iperf tutorial and opened this new Issue in the NFVO section, as you wished.

Well, at the moment I get the following Error:
**No EMS yet for host with extracted Host-ID**

I already read the other issues with similar errors/problems (No EMS yet for host with extracted Host-ID) and tried the specified hints, but no success. 

My properties are:
* Docker approach
* NFVO-Version: 4.0.0

The documentation says: 
> You should put as input for the RABBITMQ_BROKERIP the RabbitMQ IP making sure that this IP can be reached by external components (VMs, or host where will run other VNFMs) otherwise you will have runtime issues. In particular, you should select the external IP of your host on top of which the docker container is running

My Ubuntu-VM with the OpenBaton Docker-Container has the IP: 172.24.2.130
The OpenBaton Docker-Container itself has the IP 172.17.0.2
The CentOS-VM where Openstack is running has the IP: 172.24.2.150
I can ping ubuntu-vm and centos-vm in both directions

Following the documentation, I select the external IP of my host on top of which the docker container is running --> 172.24.2.130 
allocating it in the /etc/openbaton/openbaton-nfvo.properties and /etc/openbaton/openbaton-vnfm-generic.properties files, stopped and restarted the docker-container and after that I get the same error again.

So where is the mistake?!? I hope you can help me by reading the files I attached.

[generic-vnfm.log](https://github.com/openbaton/NFVO/files/1333616/generic-vnfm.log)

[openbaton-nfvo.txt](https://github.com/openbaton/NFVO/files/1333629/openbaton-nfvo.txt)

[openbaton-vnfm-generic.txt](https://github.com/openbaton/NFVO/files/1333630/openbaton-vnfm-generic.txt)

[openbaton.log](https://github.com/openbaton/NFVO/files/1333617/openbaton.log)

**Comment by gc4rella** [comment]
@mf3d3r hmm...in your config I read the following: 
`vnfm.rabbitmq.brokerIp = 172.24.2.150`
So, you are assigning the wrong IP there..as it should be 172.24.2.130. Please fix that and give it another try. Then I think should work fine.

**Comment by gc4rella** [comment]
forgot to mention the config file having that issue: openbaton-vnfm-generic.properties

**Comment by mf3d3r** [comment]
Hi, ok I'm very sorry. I mismatched the IPs in my description above.
The Ubuntu-VM with open baton docker container  has the IP: 172.24.2.150
The CentOS with open stack has the IP: 172.24.2.130

So I assigned the right IP in the files, but just described it wrong in my first comment.
 
The issue:
_No EMS yet for host with extracted Host-ID_ 
is still appearing with this configuration
vnfm.rabbitmq.brokerIp = 172.24.2.150

I renamed the files to .txt just for uploading it here.

So are there any possibilities to debug this issue. What can I do to find the error step by step. Or why there is no communication possible to EMS?

**Comment by gc4rella** [comment]
would be easier if you join on gitter (https://gitter.im/openbaton/NFVO) to troubleshoot your issue. In principle: 
* check that from the VM you can reach the rabbitmq (on port 5672, something like telnet 172.24.2.150 5672)
* check that the ems was installed correctly: please ssh into the VM and check that there is a folder /etc/openbaton/ems containing the ems conf.ini file (don't remember actual name).

**Comment by mf3d3r** [comment]
ok nice, I will do this. Just one callback. 
In your second hint, do you mean EMS is installed in the VM which is launched as an instance in openstack and I have to ssh into this VM or 
should the open baton installation contain this /etc/openbaton/ems branch with the conf.ini file?
I can't find an EMS directory in /etc/openbaton in my open baton docker container.
May be you can answer this last question and I will switch to jitter now, thanks.

**Comment by gc4rella** [comment]
You should ssh into the VM and check there if the conf.ini file is available.

**Comment by alidiab90** [comment]
@gc4rella ciao, I'm facing the same problem a time ago, I installed the EMS directory in /etc/openbaton and I think it is installed correctly, I found the conf.ini inside the EMS file found in /etc/openbaton.
the Vm instances can ping and ssh the openbaton host Ip and can reach the port 5672.
I changed the nfvo.rabbitmqIpbroker and vnfo.rabbitmqIpbroker from "localhost" to "192.168.56.104" which in the IP of the host of openbaton "192.168.56.104", for logging in the openbaton dashboard I user this IP : 192.168.56.104:8080    and then when I launch the NSD the NSR after creating the Vm instances in OpenStack and after the state goes from GRANTED to ALLOCATE , it gives me this Error:ERROR:java.lang.RuntimeException: No EMS yet for host with extracted host ID: 4523161.
